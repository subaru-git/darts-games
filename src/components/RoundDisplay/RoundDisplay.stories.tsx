import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import RoundDisplay from './RoundDisplay';

export default {
  title: 'Components/RoundDisplay',
  component: RoundDisplay,
} as ComponentMeta<typeof RoundDisplay>;

const Template: ComponentStory<typeof RoundDisplay> = (args) => <RoundDisplay {...args} />;

export const Default = Template.bind({});
Default.args = {
  count: 4,
  round: true,
};

export const Dart = Template.bind({});
Dart.args = {
  count: 1,
  round: false,
};

export const Darts = Template.bind({});
Darts.args = {
  count: 4,
  round: false,
};
