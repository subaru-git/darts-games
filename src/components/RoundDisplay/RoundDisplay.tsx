import React, { FC } from 'react';

type RoundDisplayProps = {
  count: number;
  round?: boolean;
  size?: 'sm' | 'md';
};

const RoundDisplay: FC<RoundDisplayProps> = ({ count, round, size = 'md' }) => {
  return (
    <div className='flex gap-2'>
      {round ? (
        <>
          <span className={`${size === 'md' ? 'text-2xl' : 'text-xl'} font-bold text-green-600`}>
            Round
          </span>
          <span className={`${size === 'md' ? 'text-2xl' : 'text-xl'} font-bold text-gray-500`}>
            {count}
          </span>
        </>
      ) : (
        <>
          <span className={`${size === 'md' ? 'text-2xl' : 'text-xl'} font-bold text-gray-500`}>
            {count}
          </span>
          <span
            className={`${size === 'md' ? 'text-2xl' : 'text-xl'} font-bold text-green-600`}
          >{`Dart${count > 1 ? 's' : ''}`}</span>
        </>
      )}
    </div>
  );
};

export default RoundDisplay;
