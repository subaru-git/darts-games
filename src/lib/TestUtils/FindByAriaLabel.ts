export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
