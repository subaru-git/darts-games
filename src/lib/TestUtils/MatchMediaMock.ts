import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
