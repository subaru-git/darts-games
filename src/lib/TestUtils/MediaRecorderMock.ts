export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
