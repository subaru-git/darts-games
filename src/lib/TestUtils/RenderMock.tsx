import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
