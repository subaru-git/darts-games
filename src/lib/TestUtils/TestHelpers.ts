import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}