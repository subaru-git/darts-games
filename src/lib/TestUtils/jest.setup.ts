import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
