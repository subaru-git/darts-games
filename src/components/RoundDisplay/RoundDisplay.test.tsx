import RoundDisplay from './RoundDisplay';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<RoundDisplay count={0} />);
  expect(container).toMatchSnapshot();
});
