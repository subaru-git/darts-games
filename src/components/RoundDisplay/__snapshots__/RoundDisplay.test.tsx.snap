// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="flex gap-2"
  >
    <span
      class="text-2xl font-bold text-gray-500"
    >
      0
    </span>
    <span
      class="text-2xl font-bold text-green-600"
    >
      Dart
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
